client.go

```diff
@@ -1,54 +1,118 @@
 package main
 
-import "net/http"
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// APIEndpoints maps datatype names to Ralph's API endpoints associated with them.
+var APIEndpoints = map[string]string{
+	"PhysicalHost":      "data-center-assets",
+	"VMHost":            "virtual-servers",
+	"CloudHost":         "cloud-hosts",
+	"EthernetComponent": "ethernets",
+	"BaseObject":        "base-objects",
+	// ...and so on for other data types defined for Ralph
+}
 
 // Client provides an interface to interact with Ralph via its REST API.
 type Client struct {
-	ralphURL   string
-	apiKey     string
-	apiVersion string
-	client     *http.Client
+	scannedAddr Addr
+	ralphURL    string
+	apiKey      string
+	apiVersion  string // Not used b/c Ralph doesn't have any API versioning (yet).
+	client      *http.Client
 }
 
-// New creates a new Client instance.
-func New(ralphURL, apiKey, apiVersion string, client *http.Client) *Client {
+// NewClient creates a new Client instance.
+func NewClient(ralphURL, apiKey string, scannedAddr Addr, client *http.Client) (*Client, error) {
+	// TODO(xor-xor): get rid of Query/Fragment if present
+	if apiKey == "" {
+		return nil, fmt.Errorf("API key is missing (did you forget to set it via RALPH_API_KEY environment variable?)")
+	}
+	u, err := url.Parse(ralphURL)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing Ralph's URL: %v", err)
+	}
 	return &Client{
-		ralphURL:   ralphURL,
-		apiKey:     apiKey,
-		apiVersion: apiVersion,
-		client:     http.DefaultClient,
+		scannedAddr: scannedAddr,
+		ralphURL:    u.String(),
+		apiKey:      apiKey,
+		client:      &http.Client{Timeout: time.Second * 10}, // TODO(xor-xor): This should be taken from config.
+	}, nil
+}
+
+// NewRequest creates new http.Request object initialized with headers needed for
+// communication with Ralph.
+func (c *Client) NewRequest(method, urlStr string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(method, urlStr, body)
+	if err != nil {
+		return nil, err
 	}
+	req.Header.Add("Authorization", fmt.Sprintf("Token %s", c.apiKey))
+	req.Header.Set("User-Agent", "ralph-cli")
```

Comment:
`NewRequest` is kind of "content-agnostic", b/c it is meant to be used with other functions like `SendToRalph` or `GetFromRalph`, and it is their responsibility to add such header, depending on the shape of the body that they are going to send (note that `GetFromRalph` doesn't add this header b/c there's no body; see [RFC2616 section 7.2.1](https://www.w3.org/Protocols/rfc2616/rfc2616-sec7.html#sec7.2.1) for more on that subject).